### PR TITLE
Winter cleaning

### PR DESCRIPTION
**Pull request by cthoyt**
1. Remove deprecated functions past their removal version
2. Fix some type annotations
3. Delete irrelevant/old comments

### CODE REVIEW

File: src/sssom/util.py [hunk 1]
```diff
@@ -515,11 +514,13 @@ def get_row_based_on_hierarchy(df: pd.DataFrame):
 
     :param df: Dataframe containing multiple predicates for same subject and object.
     :return: Dataframe with a single row which ranks higher in the hierarchy.
+    :raises KeyError: if no rows are available
     """
     for pred in PREDICATE_LIST:
         hierarchical_df = df[df[PREDICATE_ID] == pred]
         if not hierarchical_df.empty:
             return hierarchical_df
+    raise KeyError
```
Comment (matentzn) [inline src/sssom/util.py:523]: What is the motivation behind this change?

Comment (cthoyt) [reply src/sssom/util.py:523]: The code that uses this function always assumes it returns something. before, there was an implicit return of None at the end. If for whatever reason none of the predicates in the predicate list are available, this code should explicitly break, rather than ending up with a type error when trying to concat a None onto the dataframe that gets returned. This is why strict typing is helpful!

Comment (matentzn) [reply src/sssom/util.py:523]: Ok, thank you!

Comment (matentzn) [reply src/sssom/util.py:523]: I don't really understand what this method is doing. I guess your change is independent of that, and makes sense

**Review by matentzn — APPROVED** [review]
